src/EventListener/Doctrine/Customer/PushCustomerToMailchimp.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMailchimpPlugin\EventListener\Doctrine\Customer;
+
+use function array_key_exists;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Setono\SyliusMailchimpPlugin\Message\Command\PushCustomer;
+use Setono\SyliusMailchimpPlugin\Model\CustomerInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class PushCustomerToMailchimp
+{
+    /** @var MessageBusInterface */
+    private $messageBus;
+
+    public function __construct(MessageBusInterface $setonoSyliusMailChimpMessageBus)
+    {
+        $this->messageBus = $setonoSyliusMailChimpMessageBus;
+    }
+
+    public function postPersist(CustomerInterface $customer, LifecycleEventArgs $args): void
+    {
+        if (!$customer->isSubscribedToNewsletter()) {
+            return;
+        }
+
+        $message = new PushCustomer($customer->getId());
+        $this->messageBus->dispatch($message);
+    }
+
+    public function postUpdate(CustomerInterface $customer, LifecycleEventArgs $args): void
```

Comment:
Can you check the `get_class` of `$args`. My assumption is that it's the `PostUpdateEventArgs` class, which already has changed fields etc

Reply:
I can not find this class anywhere. Only `PreUpdateEventArgs`

Reply:
Oh yeah. Maybe better to use that event instead?

Reply:
I thought of relying on PostUpdate to be absolutely certain that the user has subscribed. And also, since we are modifying the entity `UpdatedAt/PushedToMailchimp` we would end up in an infinite loop